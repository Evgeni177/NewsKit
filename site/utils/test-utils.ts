import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
