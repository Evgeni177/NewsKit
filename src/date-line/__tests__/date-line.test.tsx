import {renderToFragmentWithTheme} from '../../test/test-utils';
import {DateLine} from '..';

const date = '2019-01-25T15:32:00';

jest.mock('date-fns/format', () => () => 'Mock Date');

describe('DateLine', () => {
  test('renders with fallback formatting', () => {
    const fragment = renderToFragmentWithTheme(DateLine, {
      date,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with provided date, format, prefix and suffix', () => {
    const fragment = renderToFragmentWithTheme(DateLine, {
      date,
      prefix: 'Updated: ',
      dateFormat: 'd MMM yyyy, H:mm',
      suffix: ', The Times',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom color', () => {
    const fragment = renderToFragmentWithTheme(DateLine, {
      date,
      color: 'inkSubtle',
    });
    expect(fragment).toMatchSnapshot();
  });
});
