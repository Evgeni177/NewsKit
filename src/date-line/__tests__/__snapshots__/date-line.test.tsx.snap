// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateLine renders with custom color 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #777777;
}

<time
    class="emotion-0"
    datetime="Mock Date"
  >
    Mock Date
  </time>
</DocumentFragment>
`;

exports[`DateLine renders with fallback formatting 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

<time
    class="emotion-0"
    datetime="Mock Date"
  >
    Mock Date
  </time>
</DocumentFragment>
`;

exports[`DateLine renders with provided date, format, prefix and suffix 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 18px;
  line-height: 1.5555555555555556;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

<time
    class="emotion-0"
    datetime="Mock Date"
  >
    Updated: Mock Date, The Times
  </time>
</DocumentFragment>
`;
